### PR TITLE
ci: e2e tests

### PR DESCRIPTION
**Pull request by orriin**
Runs the e2e tests in ci to catch regressions like #1914 before they're merged

### CODE REVIEW

**Comment by thealligatorking** [comment]
@orriin taking a look!

File: .github/workflows/e2e-subtensor-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  CARGO_TERM_COLOR: always
+  VERBOSE: ${{ github.events.input.verbose }}
+
+jobs:
+  run:
+    runs-on: SubtensorCI
+    strategy:
+      matrix:
+        rust-branch:
+          - nightly-2024-03-05
```
Comment (opendansor) [inline .github/workflows/e2e-subtensor-tests.yaml:37]: Will this line need to be updated with new releases?

Comment (orriin) [reply .github/workflows/e2e-subtensor-tests.yaml:37]: If the rust version changes, yes

**Review by opendansor — APPROVED** [review]
LGTM, one call out is to either create a PR when/if rust version updates or let us know so we can update the rust version ourselves.

**Pull request merged**